### PR TITLE
Eng 12926 Don't reverse hole loops.

### PR DESCRIPTION
**Pull request by BillWhite**
In the OGC polygon format, WKT, we represent a polygon as a counter clockwise shell followed by zero or more clockwise holes.  In S2 these are all counter clockwise, and S2 maintains the nesting relationship in a different way.  This means we need to reorder vertices in polygons to send them to S2.  In the Java GeographyValue class we represent polygons using the S2 conventions, but the constructor uses the OGC conventions.  This works alright except for the GeographyValue.add function.  This function is internal, and operates on the loops directly from the internal arrays.   Consequently, the add function does not need to reverse the order of holes.  In this commit we make sure this happens.

### CODE REVIEW

File: src/frontend/org/voltdb/types/GeographyValue.java [hunk 1]
```diff
@@ -773,15 +786,15 @@ else if (token != ',') {
      * @return  The resulting GeographyValue.
      */
     public GeographyValue add(GeographyPointValue offset) {
-        List<List<GeographyPointValue>> newLoops = new ArrayList<List<GeographyPointValue>>();
+        List<List<GeographyPointValue>> newLoops = new ArrayList<>();
```
Comment (wolffcm) [inline src/frontend/org/voltdb/types/GeographyValue.java]: I would have thought that this method is deprecated.  Should it be?

File: tests/frontend/org/voltdb/types/TestGeographyValue.java [hunk 2]
```diff
@@ -172,6 +172,7 @@ public void testGeographyValuePositive() {
         String rtStr = "POLYGON ((0.0 20.0, -17.320508076 -10.0, 17.320508076 -10.0, 0.0 20.0))";
         rtGeog = new GeographyValue(rtStr);
         assertEquals(rtStr, rtGeog.toString());
+        assertEquals(rtGeog, new GeographyValue(rtGeog.toWKT()));
```
Comment (wolffcm) [inline tests/frontend/org/voltdb/types/TestGeographyValue.java]: The original ticket made use of the `add` method.  But I don't see that you added any call to `add`.  Does this extra check actually confirm the fix?

File: src/frontend/org/voltdb/types/GeographyValue.java (hunk 1 above)
Comment (wolffcm) [reply src/frontend/org/voltdb/types/GeographyValue.java]: If there are other methods that should be deprecated, piggybacking on your fix here would be a good way to do it.

File: tests/frontend/org/voltdb/types/TestGeographyValue.java (hunk 2 above)
Comment (BillWhite) [reply tests/frontend/org/voltdb/types/TestGeographyValue.java]: I added another test.

**Pull request merged**